### PR TITLE
Net menu fixes

### PR DESCRIPTION
**Pull request by RadekVana**

### CODE REVIEW

File: src/common/selftest/selftest_esp_update.cpp [hunk 1]
```diff
@@ -297,11 +297,14 @@ bool EspCredentials::make_file() {
         return false;
     }
 
-    fputs(file_str, file.get());
+    bool failed = fputs(file_str, file.get()) == EOF;
+    if (failed) {
```
Comment (DRracer) [inline src/common/selftest/selftest_esp_update.cpp:301]: I'd recommend a more general error check ... something like
`if( fputs(...) < 0 ) ... log error`

Comment (RadekVana) [reply src/common/selftest/selftest_esp_update.cpp:301]: https://www.cplusplus.com/reference/cstdio/fputs/

Return Value
On success, a non-negative value is returned.
On error, the function returns EOF.

File: src/common/selftest/selftest_esp_update.cpp [hunk 2]
```diff
@@ -297,11 +297,14 @@ bool EspCredentials::make_file() {
...
+        log_error(Network, "ESP credentials: Unable to write into file %s", file_name);
```
Comment (DRracer) [inline src/common/selftest/selftest_esp_update.cpp:302]: you can return false in this if() branch, the file will get closed correctly

**Review by martin357 — APPROVED** [review]
Looks good, maybe some variables could be more descriptive: i.e. file_str, but that is not part of this PR.

**Pull request merged**